File: Applied Course/4.Problem Solving/12.Problems on Backtracking/4.Letter combinations of a phone number.py
# https://leetcode.com/problems/letter-combinations-of-a-phone-number/
class Solution:
    def letterCombinations(self, digits: str) -> List[str]:
        phone={
            '2':['a','b','c'],
            '3':['d','e','f'],
            '4':['g','h','i'],
            '5':['j','k','l'],
            '6':['m','n','o'],
            '7':['p','q','r','s'],
            '8':['t','u','v'],
            '9':['w','x','y','z']
        }
        def recursion(combination,digits):
            if len(digits)==0:
                result.append(combination)
            else:
                for char in phone[digits[0]]:
                    recursion(combination+char,digits[1:])
        result=[]
        if len(digits):
            recursion("",digits)
        return result

File: Applied Course/4.Problem Solving/12.Problems on Backtracking/4.Letter combinations of a phone number.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<string> letterCombinations(string digits) {
        vector<string> combos;
        vector<string> chars;
        if(digits.size()==0)
            return combos;
        combos=getChars(digits[0]);
        for(int i=1;i<digits.size();i++)
        {
            vector<string> temp=combos;
            chars=getChars(digits[i]);
            combos.clear();
            for(int j=0;j<temp.size();j++)
                for(int k=0;k<chars.size();k++)
                    combos.push_back(temp[j]+chars[k]);
        }
        return combos;
    }
    vector<string> getChars(char num)
    {
        switch(num)
        {
            case '2': return {"a","b","c"};
            case '3': return {"d","e","f"};
            case '4': return {"g","h","i"};
            case '5': return {"j","k","l"};
            case '6': return {"m","n","o"};
            case '7': return {"p","q","r","s"};
            case '8': return {"t","u","v"};
            default:  return {"w","x","y","z"};
        }
    }
};
int main()
{
    // https://leetcode.com/problems/letter-combinations-of-a-phone-number/
    return 0;
}